bundledata.go

```diff
@@ -40,6 +43,38 @@ func (lbd *legacyBundleData) setBundleData(bd *BundleData) error {
 		lbd.Applications = lbd.LegacyServices
 		lbd.unmarshaledWithServices = true
 	}
+	for appName, app := range lbd.Applications {
+		if app == nil {
+			continue
+		}
+		// Kubernetes bundles use "scale" instead of "num_units".
+		if app.Scale_ > 0 && app.NumUnits > 0 {
+			return fmt.Errorf("cannot specify both scale and num_units for application %q", appName)
+		}
+		if app.Scale_ > 0 && app.NumUnits == 0 {
+			app.NumUnits = app.Scale_
+			app.Scale_ = 0
+			lbd.Applications[appName] = app
+		}
+		// // Kubernetes bundles default to series "kubernetes".
+		if lbd.Type == kubernetes && app.Series == "" {
+			app.Series = kubernetes
+			lbd.Applications[appName] = app
+		}
+		// Non-Kubernetes bundles do not use the placement attribute.
+		if lbd.Type != kubernetes && app.Placement_ != "" {
+			return fmt.Errorf("placement (%s) not valid for non-Kubernetes application %q", app.Placement_, appName)
+		}
+		// Kubernetes bundles only use a single placement directive.
+		if app.Placement_ != "" && len(app.To) > 0 {
+			return fmt.Errorf("cannot specify both placement and to for application %q", appName)
+		}
+		if app.Placement_ != "" && len(app.To) == 0 {
```

Comment:
You know here that len(app.To) is zero, so you don't need it in the if statement.